### ISSUE TITLE
Per-page TOC links should use same anchor deduplication algo as actual anchors in page

### ISSUE THREAD

**Issue by juliusv** [prometheus/docs#403]
For example, in https://prometheus.io/docs/introduction/comparison/, half of the links are pointing at the wrong anchor.

**Comment by juliusv** [comment]
Also, on https://prometheus.io/docs/introduction/faq, the last link ("Why does Prometheus use a custom storage backend rather than [some other storage method]? Isn't the "one file per time series" approach killing performance?") is broken.

**Comment by grobie** [comment]
How is this related to the TOC? It just uses the ids defined on each heading.

**Comment by juliusv** [comment]
@grobie Ah, didn't notice that the IDs were not properly deduped either. Only the anchors themselves are! Ok, so this is not TOC-related. Will fix title.

**Comment by grobie** [comment]
I'd suggest to use the full heading path, so h2 "Prometheus vs. Graphite" -> h3 "Storage" becomes `#prometheus-vs-graphite-storage`.

**Comment by grobie** [comment]
This will break existing links, but users would still end up on the same page. This change will help to provide more descriptive links in the long run though, so I'd rather do that then extend on the `*-{counter}` model.

**Comment by juliusv** [comment]
Yeah, that'd be better. You still have to have deduping ontop of it just to be sure...

**Comment by jan--f** [comment]
Hello from the bug scrub!

At least some of the described issues here no longer reproduce, will assume this was fixed in the last years and close this accordingly. Please reopen this if there is still an issue.
